### PR TITLE
Es6 configuration

### PR DESCRIPTION
**Pull request by erikperkins**
Fixes issue #15.

Set up app to use ES6.

### CODE REVIEW

**Comment by rmacklin** [comment]
The commits https://github.com/AppFolioOnboarding/perkins_eidolon/commit/410febd449bac2f947f654ca5fcc8f5063da3134 and https://github.com/AppFolioOnboarding/perkins_eidolon/commit/f24bf470a92addb30e4f9b33672339a9cc885e69 are a bit intertwined. The first one includes "infrastructure changes" (as noted in the commit message) _as well as_ most of the work for adding the page-specific JS to the images index. It would be better to cleanly separate the two.

File: Gemfile [hunk 1]
```diff
@@ -35,6 +35,12 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 
 gem 'puma'
 
+# Gem to handle image tags
+gem 'acts-as-taggable-on'
```
Comment (rmacklin) [inline Gemfile]: This change is unrelated to ES6 configuration. Did you mean to add this gem on another branch?

File: app.json [hunk 2]
```diff
@@ -4,11 +4,20 @@
...
+    "NODE_ENV": { "required": true },
+    "RACK_ENV": { "required": true },
+    "RAILS_ENV": { "required": true },
+    "RAILS_SERVE_STATIC_FILES": { "required": true },
+    "SECRET_KEY_BASE": { "generator": "secret" }
   },
   "addons": [
-
+    "heroku-postgresql:hobby-dev",
+    "sendgrid:starter"
```
Comment (rmacklin) [inline app.json]: Where did you get this from?

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -13,4 +13,4 @@
 //= require jquery
 //= require jquery_ujs
 //= require turbolinks
-//= require_tree .
+// require_tree .
```
Comment (rmacklin) [inline app/assets/javascripts/application.js]: This line should just be removed altogether. It's a bit confusing to leave it like this.

File: config/initializers/assets.rb [hunk 4]
```diff
@@ -9,3 +9,8 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 # Rails.application.config.assets.precompile += %w( search.js )
+
+font = ['*.eot', '*.swf', '*.ttf', '*.woff', '*.woff2', '*.otf', '*.svg']
+Rails.application.config.assets.precompile += font
```
Comment (rmacklin) [inline config/initializers/assets.rb]: These changes seem unrelated to ES6 configuration. Perhaps they belong on another branch. (Additionally, you could also get fonts from a CDN instead of worrying about precompiling them and serving them ourselves.)

File: app/assets/javascripts/application.js (hunk 3 above)
Comment (erikperkins) [inline app/assets/javascripts/application.js]: Addressed in 0743f5d

File: config/initializers/assets.rb (hunk 4 above)
Comment (erikperkins) [inline config/initializers/assets.rb]: Addressed in bff3498

File: Gemfile (hunk 1 above)
Comment (erikperkins) [inline Gemfile]: Addressed in bff3498

File: app.json (hunk 2 above)
Comment (erikperkins) [inline app.json]: During the initial application setup, my app.json file was generated empty. I stole this line from the app.json file in the vanilla app.

File: Gemfile (hunk 1 above)
Comment (rmacklin) [inline Gemfile]: :+1: you can squash that change.

File: app.json (hunk 2 above)
Comment (rmacklin) [inline app.json]: I see. It'd be best not to copy things blindy :smile: - the goal is to understand every line of your changeset in a PR. Sendgrid is an email service, which we use in the Share Image story, but it's not necessary for this story, so we should remove it for now.

File: app/assets/javascripts/application.js (hunk 3 above)
Comment (rmacklin) [inline app/assets/javascripts/application.js]: :+1: you can squash that change.

File: config/initializers/assets.rb (hunk 4 above)
Comment (rmacklin) [inline config/initializers/assets.rb]: :+1: you can squash that change.

**Comment by rmacklin** [comment]
Thanks for making those changes. Let me know if you'd like to work together to squash the changes into atomic commits.

**Comment by erikperkins** [comment]
I rebased the intertwined commits.

**Comment by rmacklin** [comment]
Thanks. I left some more comments on https://github.com/AppFolioOnboarding/perkins_eidolon/commit/d628158411355f73425b8b47eef0a9fd5e2cc99d and https://github.com/AppFolioOnboarding/perkins_eidolon/commit/cb0af287f542e05ae51a7318ba7217c7ce5f59dd. I think one more rebase is in order to address those and then this'll be good to go.

**Comment by erikperkins** [comment]
I addressed your comments on d628158411355f73425b8b47eef0a9fd5e2cc99d and cb0af287f542e05ae51a7318ba7217c7ce5f59dd, and split the commit for app.json into two as you suggested.

**Comment by rmacklin** [comment]
Looks good, although it looks like the review app had a problem deploying.

**Comment by erikperkins** [comment]
@rmacklin The review app looks like it deployed successfully. Is it ready to merge now?

**Comment by rmacklin** [comment]
Yes, and then you should make sure it works on your staging application (and your production application if you then promote staging to production).
